packages/muelu/src/Utils/MueLu_UtilitiesBase_def.hpp

```diff
@@ -1012,6 +1012,9 @@ DetectDirichletRows_kokkos(const Xpetra::Matrix<SC, LO, GO, NO>& A,
     Kokkos::deep_copy(boundaryNodes2, boundaryNodes);
     return boundaryNodes2;
   }
+  // CAG: No idea why this is needed to avoid "warning: missing return statement at end of non-void function"
```

Comment:
I'm guessing this is something about the `if constexpr` not being handled properly by the compiler? is this compiler specific, or does it impact many compilers?

Reply:
I have only seen this on cuda builds.

Reply:
Yep, that's the one I tried first but it didn't get rid of the warning.